<launch>
    <arg name="name" default="imu_front" />
    <arg name="city" default="san_diego"/>
    <arg name="location" default="hotel"/>
    <arg name="device" default="/dev/imu_front" />
    <arg name="mag_LPF_bandwidth" default="15" />
    <arg name="accel_LPF_bandwidth" default="20" />
    <arg name="gyro_LPF_bandwidth" default="50" />
    <arg name="enable_iron_offset" default="false" />
    <arg name="post_IIR_LPF_bandwidth" default="17" />
    <arg name="filter_rate" default="500"/> <!-- MUST be an integer, gets passed to actual imu.launch file -->
    <arg name="imu_rate" default="1000"/> <!-- MUST be an integer, gets passed to actual imu.launch file -->

    <node pkg="riptide_hardware" name="imu_processor" type="imu_processor" output="screen" >
      <rosparam file="$(find riptide_hardware)/cfg/$(arg city).yaml" command="load"/>
      <param name="post_IIR_LPF_bandwidth" type="double" value="$(arg post_IIR_LPF_bandwidth)" />
      <param name="filter_rate" type="int" value="$(arg filter_rate)" />
    </node>

    <include file="$(find imu_3dm_gx4)/launch/imu.launch" >
      <arg name="device" value="$(arg device)" />
      <arg name="name" value="$(arg name)" />
      <arg name="city" value="$(arg city)" />
      <arg name="location" value="$(arg location)" />
      <arg name="filter_rate" value="$(arg filter_rate)" />
      <arg name="imu_rate" value="$(arg imu_rate)" />
      <arg name="mag_LPF_bandwidth" value="$(arg mag_LPF_bandwidth)" />
      <arg name="accel_LPF_bandwidth" value="$(arg accel_LPF_bandwidth)" />
      <arg name="gyro_LPF_bandwidth" value="$(arg gyro_LPF_bandwidth)" />
      <arg name="enable_iron_offset" value="$(arg enable_iron_offset)" />
    </include>
</launch>
